fdbserver/TLogServer.actor.cpp

```diff
@@ -2136,7 +2133,10 @@ Future<Void> tLogPeekMessages(PromiseType replyPromise,
 	auto messagesValue = messages.toValue();
 	reply.arena.dependsOn(messagesValue.arena());
 	reply.messages = messagesValue;
-	reply.end = clusterRecoveryVersion.present() ? clusterRecoveryVersion.get() : endVersion;
+	reply.end = endVersion;
+	if (clusterRecoveryVersion.present() && logData->stopped() && logData->version.get() < reqBegin) {
```

Comment:
> Recall the cursors peek a range (x..y] , where the lower bound x is inclusive and the upper bound y is exclusive.

That means, the cursor/peeking log is assuming that it won't receive version "y" (in reply to the peek request for range (x..y]) correct? And is the endVersion in "reply" also exclusive (and if not, isn't it incorrect that we are assigning the bumped up recovery version to the "reply.endVersion")?

If so why are we setting the bumped up value as the endVersion here?

Reply:
The cursor will keep peeking until it receives a reply with end version = y, i.e.`getEnd()`. But the _cursor_ will not include y in the reply to the caller of the cursor. 

Indeed, I believe the tLog peek code [does add 1](https://github.com/apple/foundationdb/blob/70f9440e7deb104e5903fbc8cff2e82d47fc2668/fdbserver/TLogServer.actor.cpp#L1957) to the end version in the reply.